### ISSUE TITLE
[New build - EXPERIMENTAL] 22w45a: Sensor, Shipyard & Hotfixes

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4992]
Some improvements on the sensors and shipyard side, with a bunch of lot of long overdue fixes.

The game can load once more older .mps blueprint.

Volumetric and Laser Sensor Improvement:
- Bounds detection fixed
- Can detect entities
- Can detect items
- Compatible with the event gate: can trigger for either Player / Entity, and for a given number / distance
- Have an update frequency setting.
- Can detect outside of entity bound.
- Use the latest physics and tech stack, so performance impact will be minimal.

Shipyard Improvements:
- Can rename Shipyard
- Failed request explain reason
- Can cancel request from the menu
- Various issues fixed

Community Suggestion:
#4818 Addon brick will default to the local building mode
More brick can be renamed (Rotor, hinge and shipyard)
#4553 Lower ship in 3rd person spaceship view
#4572 Sprite z-culling
#4871 #4747 Better decal targeting

Hotfixes:
#4982 Unspawning entity during box-delete can crash the game
#4555 Cannot paint decal with the steel material.
#4742 #4633 Game / Player option are not saved.
#4968 Collector issue during heavy mining.
#4769 #4670 #4678 Shipyard incorrect bound check
#2068 Thruster VFX incorrect placement.
#4923 Brick with no paint on older blueprint appear black instead of their material color
#4988 #4912 #4182 #3867 #3719 Volumetric sensor not working
#2259 Volumetric sensor in multiplayer
#2005 Sensor range not updating.
#2638 Sensor range effectively limited to entity bounds.
#2017 Sensor range is not shortened by base entity

**Comment by ProPeach** [comment]
Wow that's a big update! Has it hit Steam yet? I don't have any updates queued for experimental

**Comment by ProPeach** [comment]
What a huge update! Greate changes in there, and so many bug fixes. I bet it felt pretty good to close so many issues and clean Github up a but!

**Comment by Crimson-Artist** [comment]
awesome!

Since you're doing shipyards can we get a "Forced Exit Direction" setting? It would basically be the forced enter direction but for undocking.

**Comment by Dwarf-LordPangolin** [comment]
Thank you _so_ much for fixing the docking!

![20221111124521_1](https://user-images.githubusercontent.com/33041786/201409850-63a9b6bd-34aa-48af-aa0b-beaaf415f1bc.jpg)

**Comment by Uncle-Ulty** [comment]
I'm loving the thruster trail effect

![image](https://user-images.githubusercontent.com/73253472/201410883-54f08ad0-d44b-47a0-a821-fd6219d377df.png)

**Comment by daOyster** [comment]
So I'm definitely happy with the fixes to the volumetric sensors! One small issue I found though that isn't game breaking but can produce unexpected behaviors, I was confused why my door would trigger when I was a meter or more away from the sensor's field. Thought the bounds of the sensor wasn't applying correctly until I turned on the "View World BVH Elements" setting and saw why. I don't know completely how you are checking this, but it appears whenever the green cube from the player touches the blue sensor field it triggers. Like instead of the actual players collision box used to keep the player going through walls and the ground being used, it's stopping the check at the last Bounding volume before the players leaf node of the BVH tree is checked.

![image](https://user-images.githubusercontent.com/1314048/201486836-00bf2991-fcb4-4071-b9c2-e9ddf4609773.png)

**Comment by Arutus** [comment]
Yeah, I still can't seem to dock 100m+ ships, so whatever the issue was it has not been resolved

**Comment by Arutus** [comment]
So not sure if it's a bug, but I cannot seem to swap between multiple shipyards

**Comment by Briaireous** [comment]
Multiplayer has hella desync now

**Comment by tsunamayo** [comment]
@Briaireous Hi, is it a regression (was it working before), or a long standing issue? What happened exactly?
@Arutus No you cant choose your shipyard since the targeting update.
@daOyster I see, I will do a more precise check indeed. 
@Dwarf-LordPangolin Looking fantastique!
Thanks

**Comment by tsunamayo** [comment]
> awesome!
> 
> Since you're doing shipyards can we get a "Forced Exit Direction" setting? It would basically be the forced enter direction but for undocking.

I see, I will put it on my to-do!

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo Is it normal that it's impossible to make the Volumetric Sensors' refresh rate smaller than 1 second?

**Comment by Arutus** [comment]
then if thats the case how does one use more than one ship yard

**Comment by DrizztCode** [comment]
Hi Tsuna - awesome work with fixing the mining freezing - i just mined like crazy and it all worked perfectly - really awesome - thanks! :-)

but while i was doing so - something very very weird happened

i got it on video after the first bit - but it gets weirder as it goes along - especially at about the 2 minute mark:

https://youtu.be/9CVTchmg3Sw

if it happens again i will try and work out what makes it happen - but i was just mining and i reversed and it was just....there

and as i did things it just got weirder - ending up looking like a weird mix of the Borg Cube and the Monolith! :-)

so yeah - weird lol :-)

anyway - really awesome job with the mining stability! many thanks! :-)

(still some weird stuff with inventory flow - and i will do some proper testing and get some reports in when i can - but just so happy to have the freezing issue fixed - really amazingly great work there!)

**Comment by Briaireous** [comment]
> @Briaireous Hi, is it a regression (was it working before), or a long standing issue? What happened exactly? @Arutus No you cant choose your shipyard since the targeting update. @daOyster I see, I will do a more precise check indeed. @Dwarf-LordPangolin Looking fantastique! Thanks

The issue just started occurring after the sensor update. Pawn and twoniner can also attest to this. Dunno about anyone else, but laidai uses multiplayer quite often. 

There isn't anything specific as to what to do to recreate the issue. All you need to do is build and delete blocks. From another players perspective, the blocks do not get deleted.

**Comment by Briaireous** [comment]
Will try to get logs the next time I use multiplayer
